### ISSUE TITLE
Allow listening for `Failure`s in a `BoundedContext`

### ISSUE THREAD

**Issue by alexander-yevsyukov** [SpineEventEngine/core-java#299]
Now failures are only “travel” between client code and the backend.  But events are delivered to all interested parties.

In some cases (e.g. “Money finished”) a business logic may be interested in knowing about failures and acting accordingly.

**Comment by alexander-yevsyukov** [comment]
## Listening to failures in a Process Manager

A `ProcessManger` may fail if one of the commands that it emits fails.
We need to report the failure back to the client that originated a parent command.
What we know about a failed command is:
* Its type — this is the type of its message;
* Its ID — we have it in the `CommandContext` of the command.
What we don’t have is an ability to listen to failures. Now the failures are stored together with statuses of failed commands.

## Listening to failures

We want to use the same annotation as we have for listening to events: `@Subscribe`.
We want to have original command message and its context.

We to say: “I want to know when a `CannotDefineTask` failure happens during processing of the `DefineTask` command”. In Java this can be translated into the following:

```java
@Subscribe
void on(CannotDefineTask failure, DefineTask command, CommantContext context) {
...
}
```

**Comment by alexander-yevsyukov** [comment]
@armiol, have a look in to the suggested syntax for listening to failures.

**Comment by armiol** [comment]
@alexander-yevsyukov LGTM.

**Comment by armiol** [comment]
Resolved in #362.
